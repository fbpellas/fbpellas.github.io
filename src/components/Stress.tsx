import * as React from 'react';
import { HiPlay } from 'react-icons/hi';
import { BASE_PATH_IMG, BASE_PATH_SOUNDS } from '../constants';
import { QuizIndex } from '../types';

export interface StressProps {
  setIndexCarousel: React.Dispatch<React.SetStateAction<QuizIndex>>;
  setPageAndClear: (hash: string) => void;
}

const Stress: React.FC<StressProps> = (props) => {
  const { setIndexCarousel, setPageAndClear } = props;

  const audioPresent = new Audio(`${BASE_PATH_SOUNDS}stress/‘Present vs pre ‘sent.m4a`);
  const audioReject = new Audio(`${BASE_PATH_SOUNDS}stress/‘Reject re ‘ject.m4a`);
  const audioConduct = new Audio(`${BASE_PATH_SOUNDS}stress/‘Con duct vs Con ‘duct.m4a`);

  return (
    <div className="block-2">
      <div className="article">
        <h3 className="h3-title">Stress</h3>
        <img className="half-img" src={`${BASE_PATH_IMG}stress.jpg`} alt="Stress" />
        <div className="margin-top">
          Understanding word stress can help English language learners communicate clearly. A word stress emphasizes a
          syllable of a word with 2 or more syllables. Content words (nouns, verbs, adjectives) are usually stressed.
        </div>
        <br />
        <div>A syllable is a unit of a word with one vowel sound.</div>
        <br />
        <ul>
          <li>hi= 1 syllable</li>
          <li>flo~wer= 2 syllables</li>
          <li>e~lec~tric= 3 syllables</li>
          <li>sig~ni~fi~cant= 4 syllables</li>
        </ul>
        <br />
        <div>
          If a word has one syllable, there is no stress. Word stress is marked with the (ˈ) symbol which looks like an
          apostrophe. The stress mark comes before the stressed syllable.
        </div>
        <br />
        <p>e.g.</p>
        <ul>
          <li>hi / haɪ /</li>
          <li>ˈflower /ˈflaʊ ər/</li>
          <li>eˈlectric / ɪˈlɛk trɪk /</li>
          <li>sigˈnificant / sɪgˈnɪf ɪ kənt /</li>
        </ul>
        <br />
        <div>
          When pronouncing the stress, the syllable should sound a little higher, longer, and louder than the rest of
          the syllables.
        </div>
        <br />
        <h3 className="h3-title">Rule of Thumb</h3>
        <br />
        <div>2-syllable nouns: the first syllable is stressed</div>
        <br />
        <ul>
          <li>
            <u>per</u>son / ˈpɜr sən /
          </li>
          <li>
            <u>cas</u>tle / ˈkæs əl /
          </li>
          <li>
            <u>bas</u>ket /ˈbæs kɪt /
          </li>
        </ul>
        <br />
        <div>2-syllable verbs: the second syllable is stressed</div>
        <br />
        <ul>
          <li>
            de<u>mand</u> / dɪˈmænd /
          </li>
          <li>
            ar<u>rive</u> / əˈraɪv /
          </li>
          <li>
            com<u>plete</u> / kəmˈplit /
          </li>
        </ul>
        <br />
        <div>Some words are similar but have different meanings based on the word stress.</div>
        <br />
        <HiPlay
          className="play-icon"
          onClick={() => {
            audioPresent.play();
          }}
        />
        Play
        <ul>
          <li>ˈpresent (noun) : current moment or time</li>
          <li>preˈsent (verb) : to show</li>
        </ul>
        <br />
        <HiPlay
          className="play-icon"
          onClick={() => {
            audioReject.play();
          }}
        />
        Play
        <ul>
          <li>ˈreject (noun): something flawed or has mistakes and imperfections</li>
          <li>reˈject (verb): to refuse, to not accept</li>
        </ul>
        <br />
        <HiPlay
          className="play-icon"
          onClick={() => {
            audioConduct.play();
          }}
        />
        Play
        <ul>
          <li>ˈconduct (noun): behavior</li>
          <li>conˈduct (verb): to lead</li>
        </ul>
        <br />
        <div>What other words have similar spellings, but different meanings and stress?</div>
        <br />
        <div>Quiz yourself on how well you can distinguish word stress:</div>
        <ul>
          <li>
            <a
              className="clickable-page"
              href="#quiz"
              onClick={() => {
                setPageAndClear('quiz');
                setIndexCarousel(QuizIndex.SameWordsDifferentStress);
              }}
            >
              Same Words, Different Stress
            </a>
          </li>
          <li>
            <a
              className="clickable-page"
              href="#quiz"
              onClick={() => {
                setPageAndClear('quiz');
                setIndexCarousel(QuizIndex.WhereIsTheStress);
              }}
            >
              Where’s the Stress
            </a>
          </li>
        </ul>
      </div>
    </div>
  );
};

export { Stress };
