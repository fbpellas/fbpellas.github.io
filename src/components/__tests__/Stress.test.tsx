import React from 'react';
import { render } from '@testing-library/react';
import { Stress } from '../Stress';

test('Stress', () => {
  const fn = jest.fn();
  const { container } = render(<Stress setIndexCarousel={fn} setPageAndClear={fn} />);
  expect(container).toMatchSnapshot()
});