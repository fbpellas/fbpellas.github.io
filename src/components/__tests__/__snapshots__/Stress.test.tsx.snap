// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stress 1`] = `
<div>
  <div
    class="block-2"
  >
    <div
      class="article"
    >
      <h3
        class="h3-title"
      >
        Stress
      </h3>
      <img
        alt="Stress"
        class="half-img"
        src="https://raw.githubusercontent.com/fbpellas/fbpellas.github.io/main/public/img/stress.jpg"
      />
      <div
        class="margin-top"
      >
        Understanding word stress can help English language learners communicate clearly. A word stress emphasizes a syllable of a word with 2 or more syllables. Content words (nouns, verbs, adjectives) are usually stressed.
      </div>
      <br />
      <div>
        A syllable is a unit of a word with one vowel sound.
      </div>
      <br />
      <ul>
        <li>
          hi= 1 syllable
        </li>
        <li>
          flo~wer= 2 syllables
        </li>
        <li>
          e~lec~tric= 3 syllables
        </li>
        <li>
          sig~ni~fi~cant= 4 syllables
        </li>
      </ul>
      <br />
      <div>
        If a word has one syllable, there is no stress. Word stress is marked with the (ˈ) symbol which looks like an apostrophe. The stress mark comes before the stressed syllable.
      </div>
      <br />
      <p>
        e.g.
      </p>
      <ul>
        <li>
          hi / haɪ /
        </li>
        <li>
          ˈflower /ˈflaʊ ər/
        </li>
        <li>
          eˈlectric / ɪˈlɛk trɪk /
        </li>
        <li>
          sigˈnificant / sɪgˈnɪf ɪ kənt /
        </li>
      </ul>
      <br />
      <div>
        When pronouncing the stress, the syllable should sound a little higher, longer, and louder than the rest of the syllables.
      </div>
      <br />
      <h3
        class="h3-title"
      >
        Rule of Thumb
      </h3>
      <br />
      <div>
        2-syllable nouns: the first syllable is stressed
      </div>
      <br />
      <ul>
        <li>
          <u>
            per
          </u>
          son / ˈpɜr sən /
        </li>
        <li>
          <u>
            cas
          </u>
          tle / ˈkæs əl /
        </li>
        <li>
          <u>
            bas
          </u>
          ket /ˈbæs kɪt /
        </li>
      </ul>
      <br />
      <div>
        2-syllable verbs: the second syllable is stressed
      </div>
      <br />
      <ul>
        <li>
          de
          <u>
            mand
          </u>
           / dɪˈmænd /
        </li>
        <li>
          ar
          <u>
            rive
          </u>
           / əˈraɪv /
        </li>
        <li>
          com
          <u>
            plete
          </u>
           / kəmˈplit /
        </li>
      </ul>
      <br />
      <div>
        Some words are similar but have different meanings based on the word stress.
      </div>
      <br />
      <svg
        class="play-icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 20 20"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M10 18a8 8 0 100-16 8 8 0 000 16zM9.555 7.168A1 1 0 008 8v4a1 1 0 001.555.832l3-2a1 1 0 000-1.664l-3-2z"
          fill-rule="evenodd"
        />
      </svg>
      Play
      <ul>
        <li>
          ˈpresent (noun) : current moment or time
        </li>
        <li>
          preˈsent (verb) : to show
        </li>
      </ul>
      <br />
      <svg
        class="play-icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 20 20"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M10 18a8 8 0 100-16 8 8 0 000 16zM9.555 7.168A1 1 0 008 8v4a1 1 0 001.555.832l3-2a1 1 0 000-1.664l-3-2z"
          fill-rule="evenodd"
        />
      </svg>
      Play
      <ul>
        <li>
          ˈreject (noun): something flawed or has mistakes and imperfections
        </li>
        <li>
          reˈject (verb): to refuse, to not accept
        </li>
      </ul>
      <br />
      <svg
        class="play-icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 20 20"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M10 18a8 8 0 100-16 8 8 0 000 16zM9.555 7.168A1 1 0 008 8v4a1 1 0 001.555.832l3-2a1 1 0 000-1.664l-3-2z"
          fill-rule="evenodd"
        />
      </svg>
      Play
      <ul>
        <li>
          ˈconduct (noun): behavior
        </li>
        <li>
          conˈduct (verb): to lead
        </li>
      </ul>
      <br />
      <div>
        What other words have similar spellings, but different meanings and stress?
      </div>
      <br />
      <div>
        Quiz yourself on how well you can distinguish word stress:
      </div>
      <ul>
        <li>
          <a
            class="clickable-page"
            href="#quiz"
          >
            Same Words, Different Stress
          </a>
        </li>
        <li>
          <a
            class="clickable-page"
            href="#quiz"
          >
            Where’s the Stress
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
